scripts/quests/windurst/Carbuncle_Debacle.lua

```diff
@@ -0,0 +1,188 @@
+                        quest:setVar(player, 'Prog', 5)
+                        npcUtil.giveKeyItem(player, xi.ki.DAZE_BREAKER_CHARM)
+                    end,
+
+                    [419] = function(player, csid, option, npc)
+                        quest:complete(player)
+                    end
+                }
+            },
+
+        [xi.zone.MHAURA] =
+        {
+
+            ['Ripapa'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if questProgress == 2 then
+                        return quest:progressEvent(10022)
+                    elseif questProgress == 3 and not player:hasItem(xi.items.LIGHTNING_PENDULUM) then
+                        return quest:progressEvent(10023)
+                    end
+                end,
+            },
+
+                onEventFinish =
+                {
+                    [10022] = function(player, csid, option, npc)
+                        npcUtil.giveItem(player, xi.items.LIGHTNING_PENDULUM)
+                        quest:setVar(player, 'Prog', 3)
+                    end,
+
+                    [10023] = function(player, csid, option, npc)
+                        npcUtil.giveItem(player, xi.items.LIGHTNING_PENDULUM)
+                    end
+                }
+
+        },
+
+        [xi.zone.CLOISTER_OF_STORMS] =
+        {
+            onEventFinish =
+            {
+            [32001] = function(player, csid, option, npc)
+                local questProgress = quest:getVar(player, 'Prog')
+                    if player:getLocalVar('battlefieldWin') == 577 and questProgress == 3 then
+                        quest:setVar(player, 'Prog', 4)
+                    end
+                end
+            },
+        },
+
+        [xi.zone.RABAO] =
+        {
+            ['Agado-Pugado'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if questProgress == 5 and player:hasKeyItem(xi.ki.DAZE_BREAKER_CHARM) then
+                        return quest:progressEvent(86)
+                    elseif questProgress == 6 and not player:hasItem(xi.items.WIND_PENDULUM) then
+                        return quest:progressEvent(87)
+                    else
+                        return quest:progressEvent(88)
+                    end
+                end,
+            },
+
+                onEventFinish =
+                {
+                    [86] = function(player, csid, option, npc)
+                        npcUtil.giveItem(player, xi.items.WIND_PENDULUM)
+                        quest:setVar(player, 'Prog', 6)
+                    end,
+
+                    [87] = function(player, csid, option, npc)
+                        npcUtil.giveItem(player, xi.items.WIND_PENDULUM)
+                    end
+                },
```

Comment:
back one tab for this section and all the other onEventFinish - they should be in line with the name above.